server/src/test/java/org/elasticsearch/cluster/metadata/DesiredNodesTests.java

```diff
@@ -84,4 +95,181 @@ public void testHasSameVersion() {
         assertThat(desiredNodes.hasSameVersion(desiredNodesWithDifferentHistoryID), is(equalTo(false)));
     }
 
+    public void testClusterStateRemainsTheSameIfThereAreNoDesiredNodesDuringMembershipInfoUpgrade() {
+        final var clusterState = ClusterState.builder(new ClusterName("test")).build();
+
+        assertThat(DesiredNodes.updateDesiredNodesStatusIfNeeded(clusterState), is(sameInstance(clusterState)));
+    }
+
+    public void testDesiredNodesStatusIsUpdatedUsingCurrentClusterNodes() {
```

Comment:
nit: This looks similar to the test in `JoinTaskExecutorTests`. Can we share the code or perhaps simplify the one in `JoinTaskExecutorTests` to avoid duplication?